src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * AddCommand.java
+ * This class represents a command to add tasks to the Duke application.
+ * It handles the addition of Todo, Deadline, and Event tasks based on user input.
+ */
+
+package duke.command;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+
+import duke.DukeException;
+import duke.Ui;
+import duke.task.DateTimeUtil;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Storage;
+import duke.task.TaskList;
+import duke.task.Todo;
+
+public class AddCommand extends Command {
+    private final String fullCommand;
+
+    public AddCommand(String fullCommand) {
+        this.fullCommand = fullCommand;
+    }
+
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws DukeException, IOException {
+        String[] commandParts = fullCommand.split(" ", 2);
+        String taskType = commandParts[0];
+        String response;
+
+        if (!taskType.equals("todo") && !taskType.equals("deadline") && !taskType.equals("event")) {
+            throw new DukeException("Please enter a valid task type.");
+        }
+
+        if (commandParts.length < 2 || commandParts[1].trim().isEmpty()) {
+            throw new DukeException("The description of a " + taskType + " cannot be empty.");
+        }
+
+        String description = commandParts[1].trim();
+        switch (taskType) {
+        case "todo":
+            tasks.add(new Todo(description));
+            response = ui.showAddedTask(tasks);
+            break;
+        case "deadline":
+            String[] deadlineParts = description.split(" /by ");
+            if (deadlineParts.length < 2 || deadlineParts[1].trim().isEmpty()) {
+                throw new DukeException("The deadline time or task name is missing.");
+            }
+            tasks.add(new Deadline(deadlineParts[0].trim(), DateTimeUtil.parseDateTime(deadlineParts[1].trim())));
+            response = ui.showAddedTask(tasks);
+            break;
+        case "event":
+            String[] eventParts = description.split(" /from ", 2);
+            if (eventParts.length < 2 || eventParts[1].trim().isEmpty()) {
+                throw new DukeException("The event time details or event times are missing.");
+            }
```

Comment:
should these be extracted out to be a method on their own? To check for valid inputs